provider/consumer.py

```diff
@@ -127,9 +127,19 @@ def __init__(self, trigger, params):
         # handle the case where there may be existing triggers that do not
         # have the isJSONData field set
         if "isJSONData" in params:
-            self.parseAsJson = params["isJSONData"]
+            self.encodeValueAsJSON = params["isJSONData"]
```

Comment:
According to the Python doc for the JSON library, json.loads() technically encodes JSON strings into a Python Object. See https://docs.python.org/2/library/json.html.